collect_app/src/test/java/org/odk/collect/android/utilities/QRCodeUtilsTest.java

```diff
@@ -0,0 +1,138 @@
+package org.odk.collect.android.utilities;
+
+import android.graphics.Bitmap;
+
+import com.google.zxing.ChecksumException;
+import com.google.zxing.FormatException;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.WriterException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.application.Collect;
+import org.robolectric.RobolectricTestRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.zip.DataFormatException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.odk.collect.android.utilities.QRCodeUtils.MD5_CACHE_PATH;
+import static org.odk.collect.android.utilities.QRCodeUtils.QR_CODE_FILEPATH;
+
+
+@RunWith(RobolectricTestRunner.class)
+public class QRCodeUtilsTest {
+
+    private final File savedFile = new File(QR_CODE_FILEPATH);
```

Comment:
A more descriptive name? `savedQrCodeImage`, maybe? (Note the lowercase r in Qr. All these types (like `QRCodeUtilsTest`) should contain Qr instead of QR, but I’m not suggesting we change them now.)

Reply:
Done